deptry/dependency.py

```diff
@@ -24,12 +26,14 @@ def __init__(self, name: str, conditional: bool = False, optional: bool = False)
         self.is_conditional = conditional
         self.is_optional = optional
         self.found = self.find_metadata(name)
```

Comment:
I guess that if module_names is not `None`, there is no need to search for metadata anymore. If it does and cannot find the metadata, it will still print the warning that the module cannot be found and that e.g. `foo-example` will be defaulted to `foo_example`.

Reply:
I agree, but also think this is out of scope from this PR.

Reply:
A drawback of this though is that now it can be confusing to the user. When `python-dotenv` is in my dependencies, but not installed (so it has no metadata), with the following configuration:

```
[tool.deptry.package_module_name_map]
python-dotenv = [
    "dotenv",
]
```

a warning is still shown:

```log
Warning: Package 'python-dotenv' not found in current environment. Assuming its corresponding module name is 'python_dotenv'.
```

Since this is only used as a fallback-method, I think this is now undesired behavior? As a user I would expect that, since I manually specified which module I want the dependency to map to, I do not want it to add the default anymore.

Do you agree with the above? If not, I am okay with leaving out of scope with this PR and solving it through the separate issue you created.

Reply:
Sorry, I took a better look at the source code and I see what you mean. I agree that we should solve this in a second PR.